Determine if the command requires Docker running, fixes #3336

mikebarkas opened the pull request:
## The Problem/Issue/Bug:
For custom commands and when Docker Desktop isn't running ddev tells me "Error: unknown command 'drush' for 'ddev'  Run 'ddev --help' for usage."
(This is a Cobra error message)

## How this PR Solves The Problem:
This check that Docker is running in root init for commands.

## Manual Testing Instructions:
Turn Docker off and try running a command and a custom command.


## Automated Testing Overview:
<!-- Please provide an overview of tests introduced by this PR, or an explanation for why no tests are needed. -->
No test provided. I can provide tests with assistance.

## Related Issue Link(s):
#3336 

## Release/Deployment notes:
<!-- Does this affect anything else, or are there ramifications for other code? Does anything have to be done on deployment? -->
No


<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/3746"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

rfay commented on cmd/ddev/cmd/utils.go:
```diff
@@ -60,3 +61,27 @@ func getRequestedProjects(names []string, all bool) ([]*ddevapp.DdevApp, error)
+		"get",
+		"hostname",
+		"list",
+		"config",
```
I guess `service`  (disable and enable) would go here... I would have expected `delete` but guess it requires docker running.

mikebarkas replied:
I updated the code to always check for Docker.

rfay reviewed (commented):
I haven't tried it out yet. I had been thinking we were just trying to get shell/custom commands to behave better, but maybe this is a fine technique.

rfay commented:
FYI composer changed the output of `composer --version` in latest 2.3.x, so that's breaking TestComposerVersion. Trying to get it fixed up in https://github.com/drud/ddev/pull/3742

rfay reviewed (changes requested):
This is fine and tests fine, but let's use a more neutral language about Docker, and give a link to the docs. Docs may need an anchor for it.  So "Could not connect to a docker provider. Please start or install a docker provider. " And link to https://ddev.readthedocs.io/en/latest/users/docker_installation/

mikebarkas commented:
I will make the change to the output. Thanks.

mikebarkas commented:
Update

- Updated the output message from comment suggestion
- Added link to docker install page
- I did not feel the link needed an anchor because landing on the install page lists different operating systems

```
    Could not connect to a docker provider. Please start or install a docker provider.
    For installation help go to: https://ddev.readthedocs.io/en/latest/users/docker_installation/
```

rfay reviewed (approved):
Thanks!

rfay commented:
We forgot that `ddev config global` used to work in any directory and without docker, but I think it's OK. Broke tests in https://github.com/drud/ddev/runs/5881312332?check_suite_focus=true - but I just started Docker Desktop on that test runner and it's OK.
